programs/bpf/rust/instruction_introspection/src/lib.rs

```diff
@@ -0,0 +1,44 @@
+//! @brief Example Rust-based BPF program that exercises instruction introspection
+
+extern crate solana_sdk;
+use solana_sdk::{
+    account_info::next_account_info, account_info::AccountInfo, entrypoint,
+    entrypoint::ProgramResult, info, instruction::Instruction, program_error::ProgramError,
+    pubkey::Pubkey,
+};
+
+entrypoint!(process_instruction);
+fn process_instruction(
+    _program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    instruction_data: &[u8],
+) -> ProgramResult {
+    if instruction_data.is_empty() {
+        return Err(ProgramError::InvalidAccountData);
+    }
+
+    let secp_instruction_index = instruction_data[0];
+    let account_info_iter = &mut accounts.iter();
+    let instruction_accounts = next_account_info(account_info_iter)?;
+    assert_eq!(
+        *instruction_accounts.key,
+        solana_sdk::sysvar::instructions::id()
+    );
+    let instructions: Vec<Instruction> = unsafe {
+        bincode::deserialize(*(*instruction_accounts.data).as_ptr())
```

Comment:
We've been ignoring the sysvars, they are using bincode and this is probably a very expensive operation to deserialize.  Really need something else here.  To not tackle that issue here could you at least use  `solana_sdk::sysvar::from_account_info`?

Reply:
yea. I can use that.